autosklearn/ensemble_building/builder.py

```diff
@@ -677,41 +674,84 @@ def candidate_selection(
         for run in discarded:
             self.logger.warning(f"Have no ensemble predictions for {run}")
 
-        # Get all the ones that have a tangible loss
+        # Get all the ones that have all metrics and tangible losses
+        has_metrics = lambda r: all(m.name in r.losses for m in metrics)
+        tangible_losses = lambda r: all(loss < np.inf for loss in r.losses.values())
+
         candidates, discarded = split(
             candidates,
-            by=lambda r: r.loss < np.inf,
+            by=lambda r: has_metrics(r) and tangible_losses(r),
         )
         all_discarded.update(discarded)
+        print("all_discarded", all_discarded)
```

Comment:
This `print` should not be here I guess.

Reply:
Nope, twas some debugging, good spot :)